Playback stuck on segment with missing keyframe

### What version of Hls.js are you using?

1.4.7

### What browser (including version) are you using?

Chrome 114.0.5735.135 (Official Build) (64-bit)

### What OS (including version) are you using?

Windows 10

### Test stream

https://raw.githubusercontent.com/mattzucker/hls-test-data/main/clean-bunny/bunny_2.m3u8

### Configuration

```JavaScript
{
  "debug": true,
  "enableWorker": true,
  "lowLatencyMode": true,
  "backBufferLength": 30
}
```


### Additional player setup steps

_No response_

### Checklist

- [X] The issue observed is not already reported by searching on Github under https://github.com/video-dev/hls.js/issues
- [X] The issue occurs in the stable client (latest release) on https://hlsjs.video-dev.org/demo and not just on my page
- [X] The issue occurs in the latest client (main branch) on https://hlsjs-dev.video-dev.org/demo and not just on my page
- [X] The stream has correct Access-Control-Allow-Origin headers (CORS)
- [X] There are no network errors such as 404s in the browser console when trying to play the stream

### Steps to reproduce

1. Load the example stream.
2. Skip ahead to 5:00, around the 30th segment.
3. As the video plays it tries to add segments `sn=34` and `sn=35` to the buffer, but then evicts them immediately and re-tries.
4. When the video reaches segment 35, the playback stops.


### Expected behaviour

Buffer segments 34 and 35 once. When the video reaches segment 35, it should skip over the hole with a `bufferSeekOverHole` error and continue playing at 5:54.

### What actually happened?

Playback is stuck at the beginning of segment 35, and the segment is in an infinite loop trying to download and buffer.

### Console output

```shell
[log] > [transmuxer-interface, main]: Starting new transmux session for sn: 35 p: -1 level: 0 id: 1
        discontinuity: true
        trackSwitch: true
        contiguous: false
        accurateTimeOffset: true
        timeOffset: 350.9583333333333
        initSegmentChange: true
base-stream-controller.ts:386 [log] > [stream-controller]: Loaded fragment 35 of level 0
blob:https://hlsjs.video-dev.org/520bc58c-fef6-4f95-b9f7-9199d440f7ef:542 [log] > Debug logs enabled for "main" in hls.js version 1.4.7
transmuxer-interface.ts:379 [warn] > MPEG2-TS detected but first sync word found @ offset 104
onWorkerMessage @ transmuxer-interface.ts:379
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
transmuxer-interface.ts:379 [log] > [mp4-remuxer]: ISGenerated flag reset
transmuxer-interface.ts:379 [log] > [mp4-remuxer]: initPTS & initDTS reset
transmuxer-interface.ts:379 [log] > [mp4-remuxer]: reset next timestamp
transmuxer-interface.ts:379 [warn] > MPEG-TS PMT found at 205776 after unknown PID '256'. Backtracking to sync byte @104 to parse all TS packets.
onWorkerMessage @ transmuxer-interface.ts:379
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
transmuxer-interface.ts:379 [warn] > [mp4-remuxer]: Dropped 89 out of 269 video samples due to a missing keyframe
onWorkerMessage @ transmuxer-interface.ts:379
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
base-stream-controller.ts:1749 [log] > [stream-controller]: FRAG_LOADING->PARSING
stream-controller.ts:1264 [log] > [stream-controller]: Init video buffer, container:video/mp4, codecs[level/parsed]=[/avc1.42c01e]
base-stream-controller.ts:1749 [log] > [stream-controller]: PARSING->IDLE
base-stream-controller.ts:727 [log] > [stream-controller]: Loading fragment 35 cc: 1 of [0-59] level: 0, target: 350.958
base-stream-controller.ts:1749 [log] > [stream-controller]: IDLE->FRAG_LOADING
gap-controller.ts:229 [warn] > Playback stalling at @350.432273 due to low buffer ({"len":0.5260589999999752,"start":319.166666,"end":350.958332,"nextStart":354.666666})
_reportStall @ gap-controller.ts:229
poll @ gap-controller.ts:153
checkBuffer @ stream-controller.ts:938
onTickEnd @ stream-controller.ts:211
doTick @ stream-controller.ts:206
tick @ task-loop.ts:106
setInterval (async)
setInterval @ task-loop.ts:68
startLoad @ stream-controller.ts:127
(anonymous) @ hls.ts:402
startLoad @ hls.ts:401
filterAndSortMediaOptions @ level-controller.ts:295
onManifestLoaded @ level-controller.ts:152
emit @ index.js:203
emit @ hls.ts:287
trigger @ hls.ts:295
handleTrackOrLevelPlaylist @ playlist-loader.ts:478
onSuccess @ playlist-loader.ts:314
readystatechange @ xhr-loader.ts:231
XMLHttpRequest.send (async)
openAndSendXhr @ xhr-loader.ts:158
loadInternal @ xhr-loader.ts:117
load @ xhr-loader.ts:76
load @ playlist-loader.ts:347
onManifestLoading @ playlist-loader.ts:154
emit @ index.js:203
emit @ hls.ts:287
trigger @ hls.ts:295
loadSource @ hls.ts:389
loadSelectedStream @ main.js:372
(anonymous) @ main.js:130
dispatch @ jquery.min.js:3
r.handle @ jquery.min.js:3
Show 2 more frames
main.js:734 Error event: {type: 'mediaError', details: 'bufferStalledError', fatal: false, error: Error: Playback stalling at @350.432273 due to low buffer ({"len":0.5260589999999752,"start":319.16…, buffer: 0.5260589999999752, …}
(anonymous) @ main.js:734
emit @ index.js:203
emit @ hls.ts:287
trigger @ hls.ts:295
_reportStall @ gap-controller.ts:230
poll @ gap-controller.ts:153
checkBuffer @ stream-controller.ts:938
onTickEnd @ stream-controller.ts:211
doTick @ stream-controller.ts:206
tick @ task-loop.ts:106
setInterval (async)
setInterval @ task-loop.ts:68
startLoad @ stream-controller.ts:127
(anonymous) @ hls.ts:402
startLoad @ hls.ts:401
filterAndSortMediaOptions @ level-controller.ts:295
onManifestLoaded @ level-controller.ts:152
emit @ index.js:203
emit @ hls.ts:287
trigger @ hls.ts:295
handleTrackOrLevelPlaylist @ playlist-loader.ts:478
onSuccess @ playlist-loader.ts:314
readystatechange @ xhr-loader.ts:231
XMLHttpRequest.send (async)
openAndSendXhr @ xhr-loader.ts:158
loadInternal @ xhr-loader.ts:117
load @ xhr-loader.ts:76
load @ playlist-loader.ts:347
onManifestLoading @ playlist-loader.ts:154
emit @ index.js:203
emit @ hls.ts:287
trigger @ hls.ts:295
loadSource @ hls.ts:389
loadSelectedStream @ main.js:372
(anonymous) @ main.js:130
dispatch @ jquery.min.js:3
r.handle @ jquery.min.js:3
Show 3 more frames
transmuxer-interface.ts:85 [log] > injecting Web Worker for "main"
transmuxer-interface.ts:227 [log] > [transmuxer-interface, main]: Starting new transmux session for sn: 35 p: -1 level: 0 id: 1
        discontinuity: true
        trackSwitch: true
        contiguous: false
        accurateTimeOffset: true
        timeOffset: 350.9583333333333
        initSegmentChange: true
base-stream-controller.ts:386 [log] > [stream-controller]: Loaded fragment 35 of level 0
blob:https://hlsjs.video-dev.org/57761a54-125c-4ec0-bf7b-a763c144554c:542 [log] > Debug logs enabled for "main" in hls.js version 1.4.7
transmuxer-interface.ts:379 [warn] > MPEG2-TS detected but first sync word found @ offset 104
onWorkerMessage @ transmuxer-interface.ts:379
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
transmuxer-interface.ts:379 [log] > [mp4-remuxer]: ISGenerated flag reset
transmuxer-interface.ts:379 [log] > [mp4-remuxer]: initPTS & initDTS reset
transmuxer-interface.ts:379 [log] > [mp4-remuxer]: reset next timestamp
transmuxer-interface.ts:379 [warn] > MPEG-TS PMT found at 205776 after unknown PID '256'. Backtracking to sync byte @104 to parse all TS packets.
onWorkerMessage @ transmuxer-interface.ts:379
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
transmuxer-interface.ts:379 [warn] > [mp4-remuxer]: Dropped 89 out of 269 video samples due to a missing keyframe
onWorkerMessage @ transmuxer-interface.ts:379
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
base-stream-controller.ts:1749 [log] > [stream-controller]: FRAG_LOADING->PARSING
stream-controller.ts:1264 [log] > [stream-controller]: Init video buffer, container:video/mp4, codecs[level/parsed]=[/avc1.42c01e]
base-stream-controller.ts:1749 [log] > [stream-controller]: PARSING->IDLE
base-stream-controller.ts:727 [log] > [stream-controller]: Loading fragment 35 cc: 1 of [0-59] level: 0, target: 350.958
base-stream-controller.ts:1749 [log] > [stream-controller]: IDLE->FRAG_LOADING
transmuxer-interface.ts:85 [log] > injecting Web Worker for "main"
transmuxer-interface.ts:227 [log] > [transmuxer-interface, main]: Starting new transmux session for sn: 35 p: -1 level: 0 id: 1
        discontinuity: true
        trackSwitch: true
        contiguous: false
        accurateTimeOffset: true
        timeOffset: 350.9583333333333
        initSegmentChange: true
base-stream-controller.ts:386 [log] > [stream-controller]: Loaded fragment 35 of level 0
blob:https://hlsjs.video-dev.org/15ae700b-3de6-45ec-b3ed-8cedf64c5c76:542 [log] > Debug logs enabled for "main" in hls.js version 1.4.7
transmuxer-interface.ts:379 [warn] > MPEG2-TS detected but first sync word found @ offset 104
onWorkerMessage @ transmuxer-interface.ts:379
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
transmuxer-interface.ts:379 [log] > [mp4-remuxer]: ISGenerated flag reset
transmuxer-interface.ts:379 [log] > [mp4-remuxer]: initPTS & initDTS reset
transmuxer-interface.ts:379 [log] > [mp4-remuxer]: reset next timestamp
transmuxer-interface.ts:379 [warn] > MPEG-TS PMT found at 205776 after unknown PID '256'. Backtracking to sync byte @104 to parse all TS packets.
onWorkerMessage @ transmuxer-interface.ts:379
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
transmuxer-interface.ts:379 [warn] > [mp4-remuxer]: Dropped 89 out of 269 video samples due to a missing keyframe
onWorkerMessage @ transmuxer-interface.ts:379
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
base-stream-controller.ts:1749 [log] > [stream-controller]: FRAG_LOADING->PARSING
stream-controller.ts:1264 [log] > [stream-controller]: Init video buffer, container:video/mp4, codecs[level/parsed]=[/avc1.42c01e]
base-stream-controller.ts:1749 [log] > [stream-controller]: PARSING->IDLE
base-stream-controller.ts:727 [log] > [stream-controller]: Loading fragment 35 cc: 1 of [0-59] level: 0, target: 350.958
base-stream-controller.ts:1749 [log] > [stream-controller]: IDLE->FRAG_LOADING
transmuxer-interface.ts:85 [log] > injecting Web Worker for "main"
transmuxer-interface.ts:227 [log] > [transmuxer-interface, main]: Starting new transmux session for sn: 35 p: -1 level: 0 id: 1
        discontinuity: true
        trackSwitch: true
        contiguous: false
        accurateTimeOffset: true
        timeOffset: 350.9583333333333
        initSegmentChange: true
base-stream-controller.ts:386 [log] > [stream-controller]: Loaded fragment 35 of level 0
blob:https://hlsjs.video-dev.org/83d8d26c-a73a-4208-a899-d46d9fc90256:542 [log] > Debug logs enabled for "main" in hls.js version 1.4.7
transmuxer-interface.ts:379 [warn] > MPEG2-TS detected but first sync word found @ offset 104
onWorkerMessage @ transmuxer-interface.ts:379
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
transmuxer-interface.ts:379 [log] > [mp4-remuxer]: ISGenerated flag reset
transmuxer-interface.ts:379 [log] > [mp4-remuxer]: initPTS & initDTS reset
transmuxer-interface.ts:379 [log] > [mp4-remuxer]: reset next timestamp
transmuxer-interface.ts:379 [warn] > MPEG-TS PMT found at 205776 after unknown PID '256'. Backtracking to sync byte @104 to parse all TS packets.
onWorkerMessage @ transmuxer-interface.ts:379
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
transmuxer-interface.ts:379 [warn] > [mp4-remuxer]: Dropped 89 out of 269 video samples due to a missing keyframe
onWorkerMessage @ transmuxer-interface.ts:379
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
base-stream-controller.ts:1749 [log] > [stream-controller]: FRAG_LOADING->PARSING
stream-controller.ts:1264 [log] > [stream-controller]: Init video buffer, container:video/mp4, codecs[level/parsed]=[/avc1.42c01e]
base-stream-controller.ts:1749 [log] > [stream-controller]: PARSING->IDLE
base-stream-controller.ts:727 [log] > [stream-controller]: Loading fragment 35 cc: 1 of [0-59] level: 0, target: 350.958
base-stream-controller.ts:1749 [log] > [stream-controller]: IDLE->FRAG_LOADING
transmuxer-interface.ts:85 [log] > injecting Web Worker for "main"
transmuxer-interface.ts:227 [log] > [transmuxer-interface, main]: Starting new transmux session for sn: 35 p: -1 level: 0 id: 1
        discontinuity: true
        trackSwitch: true
        contiguous: false
        accurateTimeOffset: true
        timeOffset: 350.9583333333333
        initSegmentChange: true
base-stream-controller.ts:386 [log] > [stream-controller]: Loaded fragment 35 of level 0
blob:https://hlsjs.video-dev.org/7e5b0c82-a63e-4833-b960-3dcd5afa2e42:542 [log] > Debug logs enabled for "main" in hls.js version 1.4.7
transmuxer-interface.ts:379 [warn] > MPEG2-TS detected but first sync word found @ offset 104
onWorkerMessage @ transmuxer-interface.ts:379
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
transmuxer-interface.ts:379 [log] > [mp4-remuxer]: ISGenerated flag reset
transmuxer-interface.ts:379 [log] > [mp4-remuxer]: initPTS & initDTS reset
transmuxer-interface.ts:379 [log] > [mp4-remuxer]: reset next timestamp
transmuxer-interface.ts:379 [warn] > MPEG-TS PMT found at 205776 after unknown PID '256'. Backtracking to sync byte @104 to parse all TS packets.
onWorkerMessage @ transmuxer-interface.ts:379
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
transmuxer-interface.ts:379 [warn] > [mp4-remuxer]: Dropped 89 out of 269 video samples due to a missing keyframe
onWorkerMessage @ transmuxer-interface.ts:379
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
base-stream-controller.ts:1749 [log] > [stream-controller]: FRAG_LOADING->PARSING
stream-controller.ts:1264 [log] > [stream-controller]: Init video buffer, container:video/mp4, codecs[level/parsed]=[/avc1.42c01e]
base-stream-controller.ts:1749 [log] > [stream-controller]: PARSING->IDLE
base-stream-controller.ts:727 [log] > [stream-controller]: Loading fragment 35 cc: 1 of [0-59] level: 0, target: 350.958
base-stream-controller.ts:1749 [log] > [stream-controller]: IDLE->FRAG_LOADING
transmuxer-interface.ts:85 [log] > injecting Web Worker for "main"
transmuxer-interface.ts:227 [log] > [transmuxer-interface, main]: Starting new transmux session for sn: 35 p: -1 level: 0 id: 1
        discontinuity: true
        trackSwitch: true
        contiguous: false
        accurateTimeOffset: true
        timeOffset: 350.9583333333333
        initSegmentChange: true
base-stream-controller.ts:386 [log] > [stream-controller]: Loaded fragment 35 of level 0
blob:https://hlsjs.video-dev.org/8163a36b-ab05-4050-8a38-09a484c6308e:542 [log] > Debug logs enabled for "main" in hls.js version 1.4.7
transmuxer-interface.ts:379 [warn] > MPEG2-TS detected but first sync word found @ offset 104
onWorkerMessage @ transmuxer-interface.ts:379
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
transmuxer-interface.ts:379 [log] > [mp4-remuxer]: ISGenerated flag reset
transmuxer-interface.ts:379 [log] > [mp4-remuxer]: initPTS & initDTS reset
transmuxer-interface.ts:379 [log] > [mp4-remuxer]: reset next timestamp
transmuxer-interface.ts:379 [warn] > MPEG-TS PMT found at 205776 after unknown PID '256'. Backtracking to sync byte @104 to parse all TS packets.
onWorkerMessage @ transmuxer-interface.ts:379
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
transmuxer-interface.ts:379 [warn] > [mp4-remuxer]: Dropped 89 out of 269 video samples due to a missing keyframe
onWorkerMessage @ transmuxer-interface.ts:379
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
base-stream-controller.ts:1749 [log] > [stream-controller]: FRAG_LOADING->PARSING
stream-controller.ts:1264 [log] > [stream-controller]: Init video buffer, container:video/mp4, codecs[level/parsed]=[/avc1.42c01e]
base-stream-controller.ts:1749 [log] > [stream-controller]: PARSING->IDLE
base-stream-controller.ts:727 [log] > [stream-controller]: Loading fragment 35 cc: 1 of [0-59] level: 0, target: 350.958
base-stream-controller.ts:1749 [log] > [stream-controller]: IDLE->FRAG_LOADING
transmuxer-interface.ts:85 [log] > injecting Web Worker for "main"
transmuxer-interface.ts:227 [log] > [transmuxer-interface, main]: Starting new transmux session for sn: 35 p: -1 level: 0 id: 1
        discontinuity: true
        trackSwitch: true
        contiguous: false
        accurateTimeOffset: true
        timeOffset: 350.9583333333333
        initSegmentChange: true
base-stream-controller.ts:386 [log] > [stream-controller]: Loaded fragment 35 of level 0
blob:https://hlsjs.video-dev.org/41d1439d-c216-4e4a-84a6-70d29b27594a:542 [log] > Debug logs enabled for "main" in hls.js version 1.4.7
transmuxer-interface.ts:379 [warn] > MPEG2-TS detected but first sync word found @ offset 104
onWorkerMessage @ transmuxer-interface.ts:379
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
transmuxer-interface.ts:379 [log] > [mp4-remuxer]: ISGenerated flag reset
transmuxer-interface.ts:379 [log] > [mp4-remuxer]: initPTS & initDTS reset
transmuxer-interface.ts:379 [log] > [mp4-remuxer]: reset next timestamp
transmuxer-interface.ts:379 [warn] > MPEG-TS PMT found at 205776 after unknown PID '256'. Backtracking to sync byte @104 to parse all TS packets.
onWorkerMessage @ transmuxer-interface.ts:379
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
transmuxer-interface.ts:379 [warn] > [mp4-remuxer]: Dropped 89 out of 269 video samples due to a missing keyframe
onWorkerMessage @ transmuxer-interface.ts:379
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
base-stream-controller.ts:1749 [log] > [stream-controller]: FRAG_LOADING->PARSING
stream-controller.ts:1264 [log] > [stream-controller]: Init video buffer, container:video/mp4, codecs[level/parsed]=[/avc1.42c01e]
base-stream-controller.ts:1749 [log] > [stream-controller]: PARSING->IDLE
base-stream-controller.ts:727 [log] > [stream-controller]: Loading fragment 35 cc: 1 of [0-59] level: 0, target: 350.958
base-stream-controller.ts:1749 [log] > [stream-controller]: IDLE->FRAG_LOADING
gap-controller.ts:205 [warn] > Trying to nudge playhead over buffer-hole
_tryFixBufferStall @ gap-controller.ts:205
poll @ gap-controller.ts:164
checkBuffer @ stream-controller.ts:938
onTickEnd @ stream-controller.ts:211
doTick @ stream-controller.ts:206
tick @ task-loop.ts:106
setInterval (async)
setInterval @ task-loop.ts:68
startLoad @ stream-controller.ts:127
(anonymous) @ hls.ts:402
startLoad @ hls.ts:401
filterAndSortMediaOptions @ level-controller.ts:295
onManifestLoaded @ level-controller.ts:152
emit @ index.js:203
emit @ hls.ts:287
trigger @ hls.ts:295
handleTrackOrLevelPlaylist @ playlist-loader.ts:478
onSuccess @ playlist-loader.ts:314
readystatechange @ xhr-loader.ts:231
XMLHttpRequest.send (async)
openAndSendXhr @ xhr-loader.ts:158
loadInternal @ xhr-loader.ts:117
load @ xhr-loader.ts:76
load @ playlist-loader.ts:347
onManifestLoading @ playlist-loader.ts:154
emit @ index.js:203
emit @ hls.ts:287
trigger @ hls.ts:295
loadSource @ hls.ts:389
loadSelectedStream @ main.js:372
(anonymous) @ main.js:130
dispatch @ jquery.min.js:3
r.handle @ jquery.min.js:3
Show 2 more frames
gap-controller.ts:347 [warn] > Nudging 'currentTime' from 350.432273 to 350.53227300000003
_tryNudgeBuffer @ gap-controller.ts:347
_tryFixBufferStall @ gap-controller.ts:210
poll @ gap-controller.ts:164
checkBuffer @ stream-controller.ts:938
onTickEnd @ stream-controller.ts:211
doTick @ stream-controller.ts:206
tick @ task-loop.ts:106
setInterval (async)
setInterval @ task-loop.ts:68
startLoad @ stream-controller.ts:127
(anonymous) @ hls.ts:402
startLoad @ hls.ts:401
filterAndSortMediaOptions @ level-controller.ts:295
onManifestLoaded @ level-controller.ts:152
emit @ index.js:203
emit @ hls.ts:287
trigger @ hls.ts:295
handleTrackOrLevelPlaylist @ playlist-loader.ts:478
onSuccess @ playlist-loader.ts:314
readystatechange @ xhr-loader.ts:231
XMLHttpRequest.send (async)
openAndSendXhr @ xhr-loader.ts:158
loadInternal @ xhr-loader.ts:117
load @ xhr-loader.ts:76
load @ playlist-loader.ts:347
onManifestLoading @ playlist-loader.ts:154
emit @ index.js:203
emit @ hls.ts:287
trigger @ hls.ts:295
loadSource @ hls.ts:389
loadSelectedStream @ main.js:372
(anonymous) @ main.js:130
dispatch @ jquery.min.js:3
r.handle @ jquery.min.js:3
Show 2 more frames
main.js:734 Error event: {type: 'mediaError', details: 'bufferNudgeOnStall', error: Error: Nudging 'currentTime' from 350.432273 to 350.53227300000003
    at GapController._tryNudgeBu…, fatal: false, errorAction: {…}}
(anonymous) @ main.js:734
emit @ index.js:203
emit @ hls.ts:287
trigger @ hls.ts:295
_tryNudgeBuffer @ gap-controller.ts:349
_tryFixBufferStall @ gap-controller.ts:210
poll @ gap-controller.ts:164
checkBuffer @ stream-controller.ts:938
onTickEnd @ stream-controller.ts:211
doTick @ stream-controller.ts:206
tick @ task-loop.ts:106
setInterval (async)
setInterval @ task-loop.ts:68
startLoad @ stream-controller.ts:127
(anonymous) @ hls.ts:402
startLoad @ hls.ts:401
filterAndSortMediaOptions @ level-controller.ts:295
onManifestLoaded @ level-controller.ts:152
emit @ index.js:203
emit @ hls.ts:287
trigger @ hls.ts:295
handleTrackOrLevelPlaylist @ playlist-loader.ts:478
onSuccess @ playlist-loader.ts:314
readystatechange @ xhr-loader.ts:231
XMLHttpRequest.send (async)
openAndSendXhr @ xhr-loader.ts:158
loadInternal @ xhr-loader.ts:117
load @ xhr-loader.ts:76
load @ playlist-loader.ts:347
onManifestLoading @ playlist-loader.ts:154
emit @ index.js:203
emit @ hls.ts:287
trigger @ hls.ts:295
loadSource @ hls.ts:389
loadSelectedStream @ main.js:372
(anonymous) @ main.js:130
dispatch @ jquery.min.js:3
r.handle @ jquery.min.js:3
Show 3 more frames
base-stream-controller.ts:241 [log] > [stream-controller]: media seeking to 350.532, state: FRAG_LOADING
base-stream-controller.ts:241 [log] > [audio-stream-controller]: media seeking to 350.532, state: STOPPED
base-stream-controller.ts:241 [log] > [subtitle-stream-controller]: media seeking to 350.532, state: IDLE
stream-controller.ts:549 [log] > [stream-controller]: Media seeked to 350.532
gap-controller.ts:65 [warn] > playback not stuck anymore @350.535132, after 1ms
poll @ gap-controller.ts:65
checkBuffer @ stream-controller.ts:938
onTickEnd @ stream-controller.ts:211
doTick @ stream-controller.ts:206
tick @ task-loop.ts:106
onMediaPlaying @ stream-controller.ts:542
transmuxer-interface.ts:85 [log] > injecting Web Worker for "main"
transmuxer-interface.ts:227 [log] > [transmuxer-interface, main]: Starting new transmux session for sn: 35 p: -1 level: 0 id: 1
        discontinuity: true
        trackSwitch: true
        contiguous: false
        accurateTimeOffset: true
        timeOffset: 350.9583333333333
        initSegmentChange: true
base-stream-controller.ts:386 [log] > [stream-controller]: Loaded fragment 35 of level 0
blob:https://hlsjs.video-dev.org/8703d926-5636-4af9-9072-bcbf4b3129f4:542 [log] > Debug logs enabled for "main" in hls.js version 1.4.7
transmuxer-interface.ts:379 [warn] > MPEG2-TS detected but first sync word found @ offset 104
onWorkerMessage @ transmuxer-interface.ts:379
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
transmuxer-interface.ts:379 [log] > [mp4-remuxer]: ISGenerated flag reset
transmuxer-interface.ts:379 [log] > [mp4-remuxer]: initPTS & initDTS reset
transmuxer-interface.ts:379 [log] > [mp4-remuxer]: reset next timestamp
transmuxer-interface.ts:379 [warn] > MPEG-TS PMT found at 205776 after unknown PID '256'. Backtracking to sync byte @104 to parse all TS packets.
onWorkerMessage @ transmuxer-interface.ts:379
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
transmuxer-interface.ts:379 [warn] > [mp4-remuxer]: Dropped 89 out of 269 video samples due to a missing keyframe
onWorkerMessage @ transmuxer-interface.ts:379
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
base-stream-controller.ts:1749 [log] > [stream-controller]: FRAG_LOADING->PARSING
stream-controller.ts:1264 [log] > [stream-controller]: Init video buffer, container:video/mp4, codecs[level/parsed]=[/avc1.42c01e]
base-stream-controller.ts:1749 [log] > [stream-controller]: PARSING->IDLE
base-stream-controller.ts:727 [log] > [stream-controller]: Loading fragment 35 cc: 1 of [0-59] level: 0, target: 350.958
base-stream-controller.ts:1749 [log] > [stream-controller]: IDLE->FRAG_LOADING
transmuxer-interface.ts:85 [log] > injecting Web Worker for "main"
transmuxer-interface.ts:227 [log] > [transmuxer-interface, main]: Starting new transmux session for sn: 35 p: -1 level: 0 id: 1
        discontinuity: true
        trackSwitch: true
        contiguous: false
        accurateTimeOffset: true
        timeOffset: 350.9583333333333
        initSegmentChange: true
base-stream-controller.ts:386 [log] > [stream-controller]: Loaded fragment 35 of level 0
blob:https://hlsjs.video-dev.org/29a499fe-966a-45d4-ace9-db20626b76cd:542 [log] > Debug logs enabled for "main" in hls.js version 1.4.7
transmuxer-interface.ts:379 [warn] > MPEG2-TS detected but first sync word found @ offset 104
onWorkerMessage @ transmuxer-interface.ts:379
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
transmuxer-interface.ts:379 [log] > [mp4-remuxer]: ISGenerated flag reset
transmuxer-interface.ts:379 [log] > [mp4-remuxer]: initPTS & initDTS reset
transmuxer-interface.ts:379 [log] > [mp4-remuxer]: reset next timestamp
transmuxer-interface.ts:379 [warn] > MPEG-TS PMT found at 205776 after unknown PID '256'. Backtracking to sync byte @104 to parse all TS packets.
onWorkerMessage @ transmuxer-interface.ts:379
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
transmuxer-interface.ts:379 [warn] > [mp4-remuxer]: Dropped 89 out of 269 video samples due to a missing keyframe
onWorkerMessage @ transmuxer-interface.ts:379
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
base-stream-controller.ts:1749 [log] > [stream-controller]: FRAG_LOADING->PARSING
stream-controller.ts:1264 [log] > [stream-controller]: Init video buffer, container:video/mp4, codecs[level/parsed]=[/avc1.42c01e]
base-stream-controller.ts:1749 [log] > [stream-controller]: PARSING->IDLE
base-stream-controller.ts:727 [log] > [stream-controller]: Loading fragment 35 cc: 1 of [0-59] level: 0, target: 350.958
base-stream-controller.ts:1749 [log] > [stream-controller]: IDLE->FRAG_LOADING
transmuxer-interface.ts:85 [log] > injecting Web Worker for "main"
transmuxer-interface.ts:227 [log] > [transmuxer-interface, main]: Starting new transmux session for sn: 35 p: -1 level: 0 id: 1
        discontinuity: true
        trackSwitch: true
        contiguous: false
        accurateTimeOffset: true
        timeOffset: 350.9583333333333
        initSegmentChange: true
base-stream-controller.ts:386 [log] > [stream-controller]: Loaded fragment 35 of level 0
blob:https://hlsjs.video-dev.org/0f04eec0-6e43-4cd0-ae3a-de30d0850360:542 [log] > Debug logs enabled for "main" in hls.js version 1.4.7
transmuxer-interface.ts:379 [warn] > MPEG2-TS detected but first sync word found @ offset 104
onWorkerMessage @ transmuxer-interface.ts:379
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
transmuxer-interface.ts:379 [log] > [mp4-remuxer]: ISGenerated flag reset
transmuxer-interface.ts:379 [log] > [mp4-remuxer]: initPTS & initDTS reset
transmuxer-interface.ts:379 [log] > [mp4-remuxer]: reset next timestamp
transmuxer-interface.ts:379 [warn] > MPEG-TS PMT found at 205776 after unknown PID '256'. Backtracking to sync byte @104 to parse all TS packets.
onWorkerMessage @ transmuxer-interface.ts:379
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
transmuxer-interface.ts:379 [warn] > [mp4-remuxer]: Dropped 89 out of 269 video samples due to a missing keyframe
onWorkerMessage @ transmuxer-interface.ts:379
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
base-stream-controller.ts:1749 [log] > [stream-controller]: FRAG_LOADING->PARSING
stream-controller.ts:1264 [log] > [stream-controller]: Init video buffer, container:video/mp4, codecs[level/parsed]=[/avc1.42c01e]
gap-controller.ts:229 [warn] > Playback stalling at @350.919661 due to low buffer ({"len":0.03867099999996526,"start":319.166666,"end":350.958332,"nextStart":354.666666})
_reportStall @ gap-controller.ts:229
poll @ gap-controller.ts:153
checkBuffer @ stream-controller.ts:938
onTickEnd @ stream-controller.ts:211
doTick @ stream-controller.ts:206
tick @ task-loop.ts:106
_bufferInitSegment @ stream-controller.ts:1298
_handleTransmuxComplete @ stream-controller.ts:1085
handleTransmuxComplete @ transmuxer-interface.ts:407
onWorkerMessage @ transmuxer-interface.ts:367
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
main.js:734 Error event: {type: 'mediaError', details: 'bufferStalledError', fatal: false, error: Error: Playback stalling at @350.919661 due to low buffer ({"len":0.03867099999996526,"start":319.1…, buffer: 0.03867099999996526, …}
(anonymous) @ main.js:734
emit @ index.js:203
emit @ hls.ts:287
trigger @ hls.ts:295
_reportStall @ gap-controller.ts:230
poll @ gap-controller.ts:153
checkBuffer @ stream-controller.ts:938
onTickEnd @ stream-controller.ts:211
doTick @ stream-controller.ts:206
tick @ task-loop.ts:106
_bufferInitSegment @ stream-controller.ts:1298
_handleTransmuxComplete @ stream-controller.ts:1085
handleTransmuxComplete @ transmuxer-interface.ts:407
onWorkerMessage @ transmuxer-interface.ts:367
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
Show 1 more frame
base-stream-controller.ts:1749 [log] > [stream-controller]: PARSING->IDLE
base-stream-controller.ts:727 [log] > [stream-controller]: Loading fragment 35 cc: 1 of [0-59] level: 0, target: 350.958
base-stream-controller.ts:1749 [log] > [stream-controller]: IDLE->FRAG_LOADING
transmuxer-interface.ts:85 [log] > injecting Web Worker for "main"
transmuxer-interface.ts:227 [log] > [transmuxer-interface, main]: Starting new transmux session for sn: 35 p: -1 level: 0 id: 1
        discontinuity: true
        trackSwitch: true
        contiguous: false
        accurateTimeOffset: true
        timeOffset: 350.9583333333333
        initSegmentChange: true
base-stream-controller.ts:386 [log] > [stream-controller]: Loaded fragment 35 of level 0
blob:https://hlsjs.video-dev.org/89af3601-a795-437e-a558-c1b9fe1281d3:542 [log] > Debug logs enabled for "main" in hls.js version 1.4.7
transmuxer-interface.ts:379 [warn] > MPEG2-TS detected but first sync word found @ offset 104
onWorkerMessage @ transmuxer-interface.ts:379
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
transmuxer-interface.ts:379 [log] > [mp4-remuxer]: ISGenerated flag reset
transmuxer-interface.ts:379 [log] > [mp4-remuxer]: initPTS & initDTS reset
transmuxer-interface.ts:379 [log] > [mp4-remuxer]: reset next timestamp
transmuxer-interface.ts:379 [warn] > MPEG-TS PMT found at 205776 after unknown PID '256'. Backtracking to sync byte @104 to parse all TS packets.
onWorkerMessage @ transmuxer-interface.ts:379
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
transmuxer-interface.ts:379 [warn] > [mp4-remuxer]: Dropped 89 out of 269 video samples due to a missing keyframe
onWorkerMessage @ transmuxer-interface.ts:379
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
base-stream-controller.ts:1749 [log] > [stream-controller]: FRAG_LOADING->PARSING
stream-controller.ts:1264 [log] > [stream-controller]: Init video buffer, container:video/mp4, codecs[level/parsed]=[/avc1.42c01e]
base-stream-controller.ts:1749 [log] > [stream-controller]: PARSING->IDLE
base-stream-controller.ts:727 [log] > [stream-controller]: Loading fragment 35 cc: 1 of [0-59] level: 0, target: 350.958
base-stream-controller.ts:1749 [log] > [stream-controller]: IDLE->FRAG_LOADING
transmuxer-interface.ts:85 [log] > injecting Web Worker for "main"
transmuxer-interface.ts:227 [log] > [transmuxer-interface, main]: Starting new transmux session for sn: 35 p: -1 level: 0 id: 1
        discontinuity: true
        trackSwitch: true
        contiguous: false
        accurateTimeOffset: true
        timeOffset: 350.9583333333333
        initSegmentChange: true
base-stream-controller.ts:386 [log] > [stream-controller]: Loaded fragment 35 of level 0
blob:https://hlsjs.video-dev.org/cfaeb503-e7e4-41dd-b445-66aa3bf4eb02:542 [log] > Debug logs enabled for "main" in hls.js version 1.4.7
transmuxer-interface.ts:379 [warn] > MPEG2-TS detected but first sync word found @ offset 104
onWorkerMessage @ transmuxer-interface.ts:379
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
transmuxer-interface.ts:379 [log] > [mp4-remuxer]: ISGenerated flag reset
transmuxer-interface.ts:379 [log] > [mp4-remuxer]: initPTS & initDTS reset
transmuxer-interface.ts:379 [log] > [mp4-remuxer]: reset next timestamp
transmuxer-interface.ts:379 [warn] > MPEG-TS PMT found at 205776 after unknown PID '256'. Backtracking to sync byte @104 to parse all TS packets.
onWorkerMessage @ transmuxer-interface.ts:379
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
transmuxer-interface.ts:379 [warn] > [mp4-remuxer]: Dropped 89 out of 269 video samples due to a missing keyframe
onWorkerMessage @ transmuxer-interface.ts:379
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
base-stream-controller.ts:1749 [log] > [stream-controller]: FRAG_LOADING->PARSING
stream-controller.ts:1264 [log] > [stream-controller]: Init video buffer, container:video/mp4, codecs[level/parsed]=[/avc1.42c01e]
base-stream-controller.ts:1749 [log] > [stream-controller]: PARSING->IDLE
base-stream-controller.ts:727 [log] > [stream-controller]: Loading fragment 35 cc: 1 of [0-59] level: 0, target: 350.958
base-stream-controller.ts:1749 [log] > [stream-controller]: IDLE->FRAG_LOADING
transmuxer-interface.ts:85 [log] > injecting Web Worker for "main"
transmuxer-interface.ts:227 [log] > [transmuxer-interface, main]: Starting new transmux session for sn: 35 p: -1 level: 0 id: 1
        discontinuity: true
        trackSwitch: true
        contiguous: false
        accurateTimeOffset: true
        timeOffset: 350.9583333333333
        initSegmentChange: true
base-stream-controller.ts:386 [log] > [stream-controller]: Loaded fragment 35 of level 0
blob:https://hlsjs.video-dev.org/ec6cff7a-e685-478e-ab89-26591a185e50:542 [log] > Debug logs enabled for "main" in hls.js version 1.4.7
transmuxer-interface.ts:379 [warn] > MPEG2-TS detected but first sync word found @ offset 104
onWorkerMessage @ transmuxer-interface.ts:379
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
transmuxer-interface.ts:379 [log] > [mp4-remuxer]: ISGenerated flag reset
transmuxer-interface.ts:379 [log] > [mp4-remuxer]: initPTS & initDTS reset
transmuxer-interface.ts:379 [log] > [mp4-remuxer]: reset next timestamp
transmuxer-interface.ts:379 [warn] > MPEG-TS PMT found at 205776 after unknown PID '256'. Backtracking to sync byte @104 to parse all TS packets.
onWorkerMessage @ transmuxer-interface.ts:379
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
transmuxer-interface.ts:379 [warn] > [mp4-remuxer]: Dropped 89 out of 269 video samples due to a missing keyframe
onWorkerMessage @ transmuxer-interface.ts:379
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
base-stream-controller.ts:1749 [log] > [stream-controller]: FRAG_LOADING->PARSING
stream-controller.ts:1264 [log] > [stream-controller]: Init video buffer, container:video/mp4, codecs[level/parsed]=[/avc1.42c01e]
base-stream-controller.ts:1749 [log] > [stream-controller]: PARSING->IDLE
base-stream-controller.ts:727 [log] > [stream-controller]: Loading fragment 35 cc: 1 of [0-59] level: 0, target: 350.958
base-stream-controller.ts:1749 [log] > [stream-controller]: IDLE->FRAG_LOADING
transmuxer-interface.ts:85 [log] > injecting Web Worker for "main"
transmuxer-interface.ts:227 [log] > [transmuxer-interface, main]: Starting new transmux session for sn: 35 p: -1 level: 0 id: 1
        discontinuity: true
        trackSwitch: true
        contiguous: false
        accurateTimeOffset: true
        timeOffset: 350.9583333333333
        initSegmentChange: true
base-stream-controller.ts:386 [log] > [stream-controller]: Loaded fragment 35 of level 0
blob:https://hlsjs.video-dev.org/16946962-4091-4c92-a1a1-584c2c42de7d:542 [log] > Debug logs enabled for "main" in hls.js version 1.4.7
transmuxer-interface.ts:379 [warn] > MPEG2-TS detected but first sync word found @ offset 104
onWorkerMessage @ transmuxer-interface.ts:379
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
transmuxer-interface.ts:379 [log] > [mp4-remuxer]: ISGenerated flag reset
transmuxer-interface.ts:379 [log] > [mp4-remuxer]: initPTS & initDTS reset
transmuxer-interface.ts:379 [log] > [mp4-remuxer]: reset next timestamp
transmuxer-interface.ts:379 [warn] > MPEG-TS PMT found at 205776 after unknown PID '256'. Backtracking to sync byte @104 to parse all TS packets.
onWorkerMessage @ transmuxer-interface.ts:379
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
transmuxer-interface.ts:379 [warn] > [mp4-remuxer]: Dropped 89 out of 269 video samples due to a missing keyframe
onWorkerMessage @ transmuxer-interface.ts:379
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:88
base-stream-controller.ts:1749 [log] > [stream-controller]: FRAG_LOADING->PARSING
stream-controller.ts:1264 [log] > [stream-controller]: Init video buffer, container:video/mp4, codecs[level/parsed]=[/avc1.42c01e]
base-stream-controller.ts:1749 [log] > [stream-controller]: PARSING->IDLE
base-stream-controller.ts:727 [log] > [stream-controller]: Loading fragment 35 cc: 1 of [0-59] level: 0, target: 350.958
base-stream-controller.ts:1749 [log] > [stream-controller]: IDLE->FRAG_LOADING
```


### Chrome media internals output

_No response_

--- a/src/controller/stream-controller.ts
+++ b/src/controller/stream-controller.ts
@@ -1111,9 +1111,11 @@ export default class StreamController
     }
 
     // Avoid buffering if backtracking this fragment
-    const notFirstFragment = frag.sn !== details?.startSN;
-    if (video && remuxResult.independent !== false) {
-      if (details) {
+    if (video && details && frag.sn !== 'initSegment') {
+      const prevFrag = details.fragments[frag.sn - 1 - details.startSN];
+      const isFirstFragment = frag.sn === details.startSN;
+      const isFirstInDiscontinuity = !prevFrag || frag.cc > prevFrag.cc;
+      if (remuxResult.independent !== false) {
         const { startPTS, endPTS, startDTS, endDTS } = video;
         if (part) {
           part.elementaryStreams[video.type] = {
@@ -1123,7 +1125,12 @@ export default class StreamController
             endDTS,
           };
         } else {
-          if (video.firstKeyFrame && video.independent && chunkMeta.id === 1) {
+          if (
+            video.firstKeyFrame &&
+            video.independent &&
+            chunkMeta.id === 1 &&
+            !isFirstInDiscontinuity
+          ) {
             this.couldBacktrack = true;
           }
           if (video.dropped && video.independent) {
@@ -1137,11 +1144,15 @@ export default class StreamController
               ? video.firstKeyFramePTS
               : startPTS;
             if (
-              notFirstFragment &&
-              targetBufferTime < startTime - this.config.maxBufferHole
+              !isFirstFragment &&
+              targetBufferTime < startTime - this.config.maxBufferHole &&
+              !isFirstInDiscontinuity
             ) {
               this.backtrack(frag);
               return;
+            } else if (isFirstInDiscontinuity) {
+              // Mark segment with a gap to avoid loop loading
+              frag.gap = true;
             }
             // Set video stream start to fragment start so that truncated samples do not distort the timeline, and mark it partial
             frag.setElementaryStreamInfo(
@@ -1165,10 +1176,13 @@ export default class StreamController
           this.backtrackFragment = frag;
         }
         this.bufferFragmentData(video, frag, part, chunkMeta);
+      } else if (isFirstFragment || isFirstInDiscontinuity) {
+        // Mark segment with a gap to avoid loop loading
+        frag.gap = true;
+      } else {
+        this.backtrack(frag);
+        return;
       }
-    } else if (notFirstFragment && remuxResult.independent === false) {
-      this.backtrack(frag);
-      return;
     }
 
     if (audio) {

--- a/src/loader/fragment-loader.ts
+++ b/src/loader/fragment-loader.ts
@@ -69,8 +69,13 @@ export default class FragmentLoader {
         this.loader.destroy();
       }
       if (frag.gap) {
-        reject(createGapLoadError(frag));
-        return;
+        if (frag.tagList.some((tags) => tags[0] === 'GAP')) {
+          reject(createGapLoadError(frag));
+          return;
+        } else {
+          // Reset temporary treatment as GAP tag
+          frag.gap = false;
+        }
       }
       const loader =
         (this.loader =

